EpiAware/src/inference-methods.jl

```diff
@@ -0,0 +1,97 @@
+"""
+Run pathfinder multiple times and store the results in an array. Fails safely.
+
+# Arguments
+- `mdl::DynamicPPL.Model`: The `Turing` model to be used for inference.
```

Comment:
what has happened to templating? Going forward we should be trying to use this vs having to do catch up documentation Prs.

Reply:
This works with the templating I think? So my local help gives:

```julia-repl
help?> manypathfinder
search: manypathfinder

  manypathfinder Run multiple instances of the pathfinder algorithm and returns the pathfinder run with the largest ELBO estimate.

  Arguments
  =========

    •  mdl::DynamicPPL.Model: The model to perform inference on.

    •  nruns::Int: The number of pathfinder runs to perform.

    •  ndraws::Int: The number of draws per pathfinder run, readjusted to be at least as large as the number of chains.

    •  nchains::Int: The number of chains that will be initialised by pathfinder draws.

    •  maxiters::Int: The maximum number of optimizer iterations per pathfinder run.

    •  max_tries::Int: The maximum number of extra tries to find a valid pathfinder result.

    •  kwargs...: Additional keyword arguments passed to pathfinder.

  Returns
  =======

    •  best_pfs::PathfinderResult: Best pathfinder result by estimated ELBO.

  ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  Signatures
  ≡≡≡≡≡≡≡≡≡≡

  manypathfinder(
      mdl::DynamicPPL.Model;
      nruns,
      ndraws,
      nchains,
      maxiters,
      max_tries,
      kwargs...
  ) -> Any
  


  ────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

  Methods
  =======

  manypathfinder(
      mdl;
      nruns,
      ndraws,
      nchains,
      maxiters,
      max_tries,
      kwargs...
  )


  defined at /Users/samandfi/GitHub/CFA/Rt-without-renewal/EpiAware/src/inference-methods.jl:91
  (https://github.com/CDCgov/Rt-without-renewal/tree/957944a5eeb7d71113dd6543f8c2b3f1e39372ad//EpiAware/src/inference-methods.jl#L91).
```

Reply:
We should change the template so that the function name is on a line before the docstrings but that is a new issue?

Reply:
Actually also signature before doc string as well but again new issue

Reply:
I thought we were using the extension tools to manage the construction of the args documentation etc as you had in the original docs PR (and the other similar extension tools). It sounds like this is still in the air though so perhaps we need another discussion issue? What I mean is that yes we have the template but I thought when writing docstrings we would be automating large parts of them using the extension tools to make them less error prone (l;like the arg types).

Reply:
OK, I've made that a new issue.